webui/react/src/stores/userSettings.tsx

```diff
@@ -180,11 +180,23 @@ export class UserSettingsStore extends PollingStore {
     });
   }
 
-  /** Clears the setting, returning it to `null`. */
+  /** Clears the setting and removes the entry entirely from the database. */
   public remove(key: string): void {
-    this.updateUserSetting(key, null);
     this.#settings.update((loadable) => {
       return Loadable.map(loadable, (map) => {
+        /**
+         * Construct a dictionary to specifically target specific settings keys to clear out.
+         * To clear out the setting, `undefined` must be used instead of `null` otherwise
+         * the setting will be set to the `null` value in the database, which creates undesirable
+         * effects in UI.
+         */
+        const keys = Object.keys(map.get(key) ?? {});
+        const resetValue = keys.reduce((acc, settingsKey) => {
+          acc[settingsKey] = undefined;
```

Comment:
`undefined` doesn't serialize to JSON, `resetValue` is just `{}` here.

Reply:
It seems like what you've done is the equivalent of replacing `null` with `{}` on line 185.

Reply:
Okay, I see we're actually using the keys to iterate through in this function, just want to understand how the API server is handling this. Is it using key omission as a special case to mean "clear"?

Reply:
The issue isn't what JSON gets serialized to, the issue is that the API requires each individual settings key to be cleared. It's not designed to take in a path and clearing it out entirely. For illustration, the differences is in what `dbUpdates` end up being.

When passing in `{}` as `resetValue`, `dbUpdates` ends up being:
```
[]
```

When passing in `{ search: undefined }` as `resetValue`, `dbUpdate` ends up being:
```
[ { key: 'search', storagePath: 'task-list-global', value: undefined } ]
```

Reply:
> Okay, I see we're actually using the keys to iterate through in this function, just want to understand how the API server is handling this. Is it using key omission as a special case to mean "clear"?

Yeah unfortunately key omission as meaning `clear` isn't the current behavior, but also not sure if we want such an implicit behavior. Might accidentally clear out settings via key omission if we're not careful. It might be useful to have a separate clear API call to remove everything under a particular `storagePath`.

Reply:
Ok, after delving through the code this looks mostly correct with the undefined eventually being coerced to an empty string which gets checked for on the API server but _only_ for settings that are objects. You need to check if the value we're clearing is an object before this logic and preserve the old logic otherwise.